build.sh

```diff
@@ -230,23 +367,47 @@ __BinDir="$__rootbinpath/$__BuildOS.$__BuildArch.$__BuildType/Native"
 
 setup_dirs
 
-# Check prereqs.
+if $__buildmanaged; then
 
-check_managed_prereqs
+    # Prepare the system
 
-# Prepare the system
+    prepare_managed_build
 
-prepare_managed_build
+    # Build the corefx native components.
 
-# Build the wcf managed components.
+    build_managed_corefx
 
-build_managed_wcf
-
-# Build complete
+    # Build complete
+fi
 
 # If managed build failed, exit with the status code of the managed build
 if [ $BUILDERRORLEVEL != 0 ]; then
     exit $BUILDERRORLEVEL
 fi
 
+# Configure environment if we are doing a cross compile.
+if [ "$__CrossBuild" == 1 ]; then
+    export CROSSCOMPILE=1
+    if ! [[ -n "$ROOTFS_DIR" ]]; then
+        export ROOTFS_DIR="$__scriptpath/cross/rootfs/$__BuildArch"
+    fi
```

Comment:
We'll need to test this if we keep it.  That means building with -cross and verifying both with and without ROOTFS_DIR set.  This testing could be done after the merge, providing a normal TFS build succeeds.